client/app/(auth)/signUp.tsx

```diff
@@ -0,0 +1,283 @@
+            render={({ field: { onChange, value } }) => (
+              <View
+                className={`mb-1 min-h-14 flex-row items-center justify-between rounded-xl border ${errors.password || errorMessage ? "border-red-600" : "border-gray-920"} px-3 py-2`}
+              >
+                <TextInput
+                  className="flex-1 text-base-white placeholder:pbk-b1"
+                  onChangeText={(text) => {
+                    onChange(text);
+                    setErrorMessage("");
+                  }}
+                  placeholder="Enter password"
+                  placeholderTextColor="gray"
+                  secureTextEntry={hidePassword}
+                  value={value}
+                />
+                <Pressable
+                  className="ml-2 flex-row gap-2"
+                  onPress={() => setHidePassword(!hidePassword)}
+                >
+                  {hidePassword ? (
+                    <EyeSlash color="gray" size={20} weight="bold" />
+                  ) : (
+                    <Eye color="gray" size={20} weight="bold" />
+                  )}
+                  {(errors.password || errorMessage) && (
+                    <WarningCircle color="#dc2626" size={20} weight="bold" />
+                  )}
+                </Pressable>
+              </View>
+            )}
+          />
+          <View className="mb-5">
+            <Text
+              className={`pbk-b3 ${errors.password || errorMessage ? "text-red-600" : "text-gray-600"}`}
+            >
+              {errors.password
+                ? errors.password.message
+                : "Password must be at least 8 characters"}
+            </Text>
+            {errorMessage && (
+              <Text className="pbk-b3 mb-4 text-red-600">{errorMessage}</Text>
+            )}
+          </View>
+          <Pressable
+            className={`${!isValid ? "bg-purple-900" : "bg-purple-600"} min-h-12 w-full items-center justify-center rounded-md`}
+            disabled={!isValid}
+            onPress={handleSubmit(signUpUser)}
+          >
+            {loading ? (
+              <Spinner />
+            ) : (
+              <Text
+                className={`pbk-h6 text-center ${!isValid ? "text-gray-400" : "text-base-white"}`}
+              >
+                SIGN UP
+              </Text>
+            )}
+          </Pressable>
+
+          <View className="pbk-b1 my-5 flex-row flex-wrap">
+            <Text className="text-gray-600">
+              By signing up, you agree to our
+            </Text>
+            <Pressable onPress={() => router.back()}>
+              <Text className="text-purple-600"> Terms of Service</Text>
+            </Pressable>
+            <Text className="text-gray-600"> and </Text>
+            <Pressable onPress={() => router.back()}>
+              <Text className="text-purple-600">Privacy Policy</Text>
+            </Pressable>
+            <Text className="text-gray-600">.</Text>
```

Comment:
A few things going on here. I think the better way to style this might be to wrap the entire sentence in a text block and then have nested text blocks, similar to p and span tags on web. The markup reads a bit easier and more importantly, the spaces will take into account line breaks if we do this. For example, on my phone there's a weird gap between the beginning of the line and the "and" because my screen size is smaller

The other thing is that we can probably get away with only using a `Text` tag an putting the `onPress` on that for links. Honestly I'm not sure when we should use this over text buttons, maybe it should be a style thing like the text buttons have more padding and specific fonts?

![IMG_4413](https://github.com/user-attachments/assets/70d73345-dc5e-43ad-a560-1149c183f086)

Reply:
Cool thanks for bringing this up. I wasn't sure what was best and turns out there were a few other things wrong with it too. I removed the View component and wrapped everything inside a larger Text component and removed the Pressable's. I replaced them with in-line Text components as suggested with an onPress function. I feel like if this had more styling I would try to use a text button like you mentioned, but this seemed simple enough that I don't think warrants any additional component.

Let me know if you are happy with the changes.

This is what I am going with: 


```suggestion
        <Text className="pbk-b1 my-5 flex-row flex-wrap text-gray-600">
            By signing up, you agree to our{" "}
            <Text className="text-purple-600">Terms of Service</Text> and{" "}
            <Text className="text-purple-600">Privacy Policy</Text>.
        </Text>
```

Reply:
Nice, this look better on my screen and the markup looks cleaner too 👍